File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0, r = height.size()-1, maxArea = 0;
        while (l<r){
            int area = (r-l)*min(height[l], height[r]);
            maxArea = max(maxArea, area);
            if (height[l]<height[r]) l++;
            else r--;
        }
        return maxArea;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let p1 = 0, p2 = height.length-1, curArea=0, area = 0;
    
    while (p1<p2){
        curArea = Math.min(height[p1], height[p2])*(p2-p1);
        area = Math.max(area, curArea);
        
        if (height[p1]<=height[p2]) p1++;
        else p2--;
    }
    return area;
};